quantum/pointing_device/pointing_device_auto_mouse.c

```diff
@@ -334,14 +345,18 @@ bool process_auto_mouse(uint16_t keycode, keyrecord_t* record) {
     if (!(AUTO_MOUSE_ENABLED)) return true;
 
     switch (keycode) {
-        // Skip Mod keys to avoid layer reset
+        // Skip Mod keys, KC_NO, and layer lock to avoid layer reset
+        case KC_NO:
         case KC_LEFT_CTRL ... KC_RIGHT_GUI:
         case QK_MODS ... QK_MODS_MAX:
+        case QK_LLCK:
             break;
         // TO((AUTO_MOUSE_TARGET_LAYER))-------------------------------------------------------------------------------
         case QK_TO ... QK_TO_MAX:
             if (QK_TO_GET_LAYER(keycode) == (AUTO_MOUSE_TARGET_LAYER)) {
                 if (!(record->event.pressed)) auto_mouse_toggle();
+            } else {
+                auto_mouse_context.status.is_toggled = false;
```

Comment:
Can we add some context added on why this `else` is needed. Why its not using something like `auto_mouse_reset`, and why the other layer keycodes do not need this handling.